python/pyspark/sql/types.py

```diff
@@ -764,6 +764,21 @@ def __eq__(self, other):
         return type(self) == type(other)
 
 
+def _is_datatype_with_udt(dt):
```

Comment:
`_has_udt()` for short?

Reply:
Is `MapType` explicitly unsupported? if so, should we add document to this method?

Reply:
fixed